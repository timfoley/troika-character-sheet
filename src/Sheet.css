  .sheet {
      display              : grid;
      grid-gap             : 12px;
      width                : 100%;
      grid-template-columns: 1fr;
      grid-template-areas  :
          "left"
          "right";
  }

  .left-grid {
      grid-area: left;
      display  : grid;
      grid-template-areas:
          "basics"
          "stats"
          "skills";
  }



  .right-grid {
      grid-area: right;
      display  : grid;
      grid-gap : 8px;
      grid-template-areas:
          "weapons   weapons"
          "inventory inventory"
          "wearing   wearing"
          "money     provisions";
  }

  @media (min-width: 500px) {
      .left-grid {
          grid-template-columns: 1fr 1fr 1fr 110px;
          grid-template-areas  :
              "basics basics basics basics"
              "skills skills skills stats";
      }

      .right-grid {
          grid-template-areas:
              "weapons   weapons   weapons   weapons"
              "inventory inventory inventory inventory"
              "wearing   wearing   wearing   money"
              "wearing   wearing   wearing   provisions";
      }

      .sheet {
          padding: 20px;
          border : 1px solid var(--primary-dark);
      }
  }

  @media (min-width: 800px) {
      .sheet {
          grid-template-areas:
              "left  right ";
          grid-template-columns: 1fr 1fr;
      }
  }