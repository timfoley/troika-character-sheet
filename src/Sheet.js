import React from "react";
import PropTypes from "prop-types";

import "./Sheet.css";

import { Basics } from "./Basics";
import { Stats } from "./Stats";
import { Skills } from "./Skills";
import { Weapons } from "./Weapons";
import { Inventory } from "./Inventory";
import { OtherGear } from "./OtherGear";

export function Sheet(props) {
	const updateCharacter = (field, value) => {
		props.setCharacter({ ...props.character, [field]: value });
	};

	return (
		<div className="sheet">
			<div className="left-grid">
				<Basics
					name={props.character.name}
					background={props.character.background}
					special={props.character.special}
					updateCharacter={updateCharacter}
				/>
				<Stats
					stamina={props.character.stamina}
					skill={props.character.skill}
					luck={props.character.luck}
					updateCharacter={updateCharacter}
				/>
				<Skills
					skills={props.character.skills}
					skillStat={props.character.skill}
				/>
			</div>
			<div className="right-grid">
				<Weapons weapons={props.character.weapons} />
				<Inventory inventory={props.character.inventory} />
				<OtherGear
					wearing={props.character.wearing}
					money={props.character.money}
					provisions={props.character.provisions}
				/>
			</div>
		</div>
	);
}

Sheet.propTypes = {
	character: PropTypes.object,
	setCharacter: PropTypes.func
};
