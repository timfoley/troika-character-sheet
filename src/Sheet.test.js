import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";

import { Sheet } from "./Sheet";

it("renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<Sheet />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
	const { container } = render(<Sheet />);
	expect(container.firstChild).toMatchSnapshot();
});
