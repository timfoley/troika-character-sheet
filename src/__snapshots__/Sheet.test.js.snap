// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="sheet"
>
  <div
    class="left-grid"
  >
    <div
      class="basics"
    >
      <label
        class="basics__name"
      >
        Name
        <input
          class="basics__input basics__input--name"
          type="text"
          value=""
        />
      </label>
      <label
        class="basics__background"
      >
        Background
        <input
          class="basics__input basics__input--background "
          type="text"
          value=""
        />
      </label>
      <label
        class="basics__special"
      >
        Special
        <textarea
          class="basics__input basics__input--special"
          rows="4"
        />
      </label>
    </div>
    <div
      class="stats"
    >
      <label
        class="stat skill"
      >
        Skill
        <input
          class="skill-stat stat__input stat__input--main"
          type="number"
          value="0"
        />
      </label>
      <div
        class="stat stamina"
      >
        <label>
          Stamina
          <input
            class="stamina-remaining stat__input stat__input--main"
            max="0"
            type="number"
            value="0"
          />
        </label>
        <label
          class="sub-label"
        >
          <input
            class="stamina-max stat__input stat__input--sub"
            type="number"
            value="0"
          />
          Max
        </label>
      </div>
      <div
        class="stat luck"
      >
        <label>
          Luck
          <input
            class="luck-remaining stat__input stat__input--main"
            max="0"
            type="number"
            value="0"
          />
        </label>
        <label
          class="sub-label"
        >
          <input
            class="luck-max stat__input stat__input--sub"
            type="number"
            value="0"
          />
          Max
        </label>
      </div>
    </div>
    <div
      class="skills"
    >
      <h2>
        Advanced Skills and Spells
      </h2>
      <div
        class="skills__grid"
      >
        <div
          class="skills__grid__heading skill__box--rank"
        >
          Rank
        </div>
        <div
          class="skills__grid__heading skill__box--skill"
        >
          Skill
        </div>
        <div
          class="skills__grid__heading skill__box--total"
        >
          Total
        </div>
        <div
          class="skill__name"
        >
          <input
            aria-label="skill name"
            class="skill__name__input"
            type="text"
            value=""
          />
        </div>
        <input
          aria-label="skill rank"
          class="skill__box skill__box--rank hide-spinners zero-to-hide"
          type="number"
          value="0"
        />
        <div
          aria-label="skill stat"
          class="skill__box skill__box--skill-stat zero-to-hide"
        >
          0
        </div>
        <div
          aria-label="skill total"
          class="skill__box skill__box--total zero-to-hide"
        >
          0
        </div>
        <div
          class="skill__name"
        >
          <input
            aria-label="skill name"
            class="skill__name__input"
            type="text"
            value=""
          />
        </div>
        <input
          aria-label="skill rank"
          class="skill__box skill__box--rank hide-spinners zero-to-hide"
          type="number"
          value="0"
        />
        <div
          aria-label="skill stat"
          class="skill__box skill__box--skill-stat zero-to-hide"
        >
          0
        </div>
        <div
          aria-label="skill total"
          class="skill__box skill__box--total zero-to-hide"
        >
          0
        </div>
        <div
          class="skill__name"
        >
          <input
            aria-label="skill name"
            class="skill__name__input"
            type="text"
            value=""
          />
        </div>
        <input
          aria-label="skill rank"
          class="skill__box skill__box--rank hide-spinners zero-to-hide"
          type="number"
          value="0"
        />
        <div
          aria-label="skill stat"
          class="skill__box skill__box--skill-stat zero-to-hide"
        >
          0
        </div>
        <div
          aria-label="skill total"
          class="skill__box skill__box--total zero-to-hide"
        >
          0
        </div>
        <div
          class="skill__name"
        >
          <input
            aria-label="skill name"
            class="skill__name__input"
            type="text"
            value=""
          />
        </div>
        <input
          aria-label="skill rank"
          class="skill__box skill__box--rank hide-spinners zero-to-hide"
          type="number"
          value="0"
        />
        <div
          aria-label="skill stat"
          class="skill__box skill__box--skill-stat zero-to-hide"
        >
          0
        </div>
        <div
          aria-label="skill total"
          class="skill__box skill__box--total zero-to-hide"
        >
          0
        </div>
        <div
          class="skill__name"
        >
          <input
            aria-label="skill name"
            class="skill__name__input"
            type="text"
            value=""
          />
        </div>
        <input
          aria-label="skill rank"
          class="skill__box skill__box--rank hide-spinners zero-to-hide"
          type="number"
          value="0"
        />
        <div
          aria-label="skill stat"
          class="skill__box skill__box--skill-stat zero-to-hide"
        >
          0
        </div>
        <div
          aria-label="skill total"
          class="skill__box skill__box--total zero-to-hide"
        >
          0
        </div>
      </div>
      <div
        area-label="click to add a skill"
        class="skills__add-button clickable"
        tabindex="0"
      >
        Add Skill
      </div>
    </div>
  </div>
  <div
    class="right-grid"
  >
    <div
      class="weapons"
    >
      <h2>
        Weapons
      </h2>
      <div
        class="weapons__grid"
      >
        <div
          class="weapon__name"
        >
          <input
            aria-label="weapon name"
            class="weapon__name__input"
            type="text"
            value=""
          />
        </div>
        <div
          class="weapon__damage"
        >
          <input
            aria-label="1 damage value for "
            class="weapon__damage__input weapon__damage__input--0 hide-spinners"
            type="number"
            value="0"
          />
          <div
            aria-hidden="true"
            class="weapon__damage__roll"
          >
            1
          </div>
        </div>
        <div
          class="weapon__damage"
        >
          <input
            aria-label="2 damage value for "
            class="weapon__damage__input weapon__damage__input--1 hide-spinners"
            type="number"
            value="0"
          />
          <div
            aria-hidden="true"
            class="weapon__damage__roll"
          >
            2
          </div>
        </div>
        <div
          class="weapon__damage"
        >
          <input
            aria-label="3 damage value for "
            class="weapon__damage__input weapon__damage__input--2 hide-spinners"
            type="number"
            value="0"
          />
          <div
            aria-hidden="true"
            class="weapon__damage__roll"
          >
            3
          </div>
        </div>
        <div
          class="weapon__damage"
        >
          <input
            aria-label="4 damage value for "
            class="weapon__damage__input weapon__damage__input--3 hide-spinners"
            type="number"
            value="0"
          />
          <div
            aria-hidden="true"
            class="weapon__damage__roll"
          >
            4
          </div>
        </div>
        <div
          class="weapon__damage"
        >
          <input
            aria-label="5 damage value for "
            class="weapon__damage__input weapon__damage__input--4 hide-spinners"
            type="number"
            value="0"
          />
          <div
            aria-hidden="true"
            class="weapon__damage__roll"
          >
            5
          </div>
        </div>
        <div
          class="weapon__damage"
        >
          <input
            aria-label="6 damage value for "
            class="weapon__damage__input weapon__damage__input--5 hide-spinners"
            type="number"
            value="0"
          />
          <div
            aria-hidden="true"
            class="weapon__damage__roll"
          >
            6
          </div>
        </div>
        <div
          class="weapon__damage"
        >
          <input
            aria-label="7+ damage value for "
            class="weapon__damage__input weapon__damage__input--6 hide-spinners"
            type="number"
            value="0"
          />
          <div
            aria-hidden="true"
            class="weapon__damage__roll"
          >
            7+
          </div>
        </div>
      </div>
      <div
        class="weapons__add-button clickable"
        tabindex="0"
      >
        Add Weapon
      </div>
    </div>
    <div
      class="inventory"
    >
      <h2>
        Inventory
      </h2>
      <ol
        class="inventory__grid"
      >
        <li
          class="item"
        >
          <textarea
            aria-label="Slot 1"
            class="item__input"
            id="0"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 2"
            class="item__input"
            id="1"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 3"
            class="item__input"
            id="2"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 4"
            class="item__input"
            id="3"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 5"
            class="item__input"
            id="4"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 6"
            class="item__input"
            id="5"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 7"
            class="item__input"
            id="6"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 8"
            class="item__input"
            id="7"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 9"
            class="item__input"
            id="8"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 10"
            class="item__input"
            id="9"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 11"
            class="item__input"
            id="10"
            type="text"
          />
        </li>
        <li
          class="item blank"
        >
          <textarea
            aria-label="Empty slot 12"
            class="item__input"
            id="11"
            type="text"
          />
        </li>
      </ol>
    </div>
    <label
      class="gear__label wearing"
    >
      Wearing
      <textarea
        class="gear__input"
      />
    </label>
    <label
      class="gear__label money box"
    >
      Money
      <input
        class="gear__input"
        type="number"
        value="0"
      />
    </label>
    <label
      class="gear__label provisions box"
    >
      Provisions
      <input
        class="gear__input"
        type="number"
        value="0"
      />
    </label>
  </div>
</div>
`;
